pkg/manager/internal.go

```diff
@@ -313,7 +336,46 @@ func (cm *controllerManager) startLeaderElectionRunnables() {
 		// Write any Start errors to a channel so we can return them
 		ctrl := c
 		go func() {
-			cm.errChan <- ctrl.Start(cm.internalStop)
+			if leRunnable, ok := ctrl.(LeaderElectionRunnable); ok {
+				leID := leRunnable.GetID()
+
+				// Check that leader election ID is defined
+				if leID == "" {
+					cm.errChan <- errors.New("LeaderElectionID must be configured")
+					return
+				}
+
+				// Check that leader election ID is unique
+				if _, exists := cm.resourceLocksMap[leID]; exists {
+					cm.errChan <- errors.New("LeaderElectionID must be unique")
```

Comment:
that's not entirely true -- a group of controllers could choose to run under the same leader election

Reply:
(which might make sense if they needed the same resources, etc.